doc/sql.extensions/README.profiler.md

```diff
@@ -0,0 +1,164 @@
+
+## Procedure `FINISH_SESSION`
+
+`RDB$PROFILER.FINISH_SESSION` finishes the current profiler session.
+
+If `UPDATE_SNAPSHOT` is `TRUE` the snapshot tables are updated with data of the finished session (and old finished sessions not yet present in the snapshot). Otherwise data remains only in the engine memory for later update.
+
+Calling `RDB$PROFILER.FINISH_SESSION(TRUE)` has the same semantics of calling `RDB$PROFILER.FINISH_SESSION(FALSE)` followed by `RDB$PROFILER.UPDATE_SNAPSHOT`.
+
+Input parameters:
+ - `UPDATE_SNAPSHOT` type `BOOLEAN NOT NULL`
+
+## Procedure `UPDATE_SNAPSHOT`
+
+`RDB$PROFILER.UPDATE_SNAPSHOT` updates the system tables snapshots with data from the profile sessions in memory.
+
+After update data is stored in tables `RDB$PROFILE_SESSIONS`, `RDB$PROFILE_REQUESTS` and `RDB$PROFILE_STATS` and may be read and analyzed by the user.
+
+It also removes finished sessions from engine memory, so if `RDB$PROFILER.PURGE_SNAPSHOTS` is later called these data are not recovered.
+
+## Procedure `PURGE_SNAPSHOTS`
+
+`RDB$PROFILER.PURGE_SNAPSHOTS` removes all profile snapshots from the system tables and remove finished profile sessions from engine memory.
+
+# Snapshot system tables
+
+The profile snaphsot tables are like user's global temporary table with `ON COMMIT PRESERVE ROWS`. Data are per attachment and remains there until disconnected or explicitely purged with `RDB$PROFILER.PURGE_SNAPSHOTS`.
+
+## Table `RDB$PROFILE_SESSIONS`
+
+ - `RDB$PROFILE_SESSION_ID` type `BIGINT` - Profile session ID
+ - `RDB$TIMESTAMP` type `TIMESTAMP WITH TIME ZONE` - Moment the profile session was started
+
+## Table `RDB$PROFILE_REQUESTS`
+
+ - `RDB$PROFILE_SESSION_ID` type `BIGINT` - Profile session ID
+ - `RDB$PROFILE_REQUEST_ID` type `BIGINT` - Request ID
+ - `RDB$TIMESTAMP` type `TIMESTAMP WITH TIME ZONE` - Moment this request was first gathered profile data
+ - `RDB$REQUEST_TYPE` type `VARCHAR(20) CHARACTER SET UTF8` - BLOCK, FUNCTION, PROCEDURE or TRIGGER
+ - `RDB$PACKAGE_NAME` type `CHAR(63) CHARACTER SET UTF8` - Package of FUNCTION or PROCEDURE
+ - `RDB$ROUTINE_NAME` type `CHAR(63) CHARACTER SET UTF8` - Routine name of FUNCTION, PROCEDURE or TRIGGER
+ - `RDB$SQL_TEXT` type `BLOB subtype TEXT CHARACTER SET UTF8` - SQL text for BLOCK
+
+## Table `RDB$PROFILE_STATS`
+
+ - `RDB$PROFILE_SESSION_ID` type `BIGINT` - Profile session ID
+ - `RDB$PROFILE_REQUEST_ID` type `BIGINT` - Request ID
+ - `RDB$LINE` type `INTEGER` - Line number of the statement
+ - `RDB$COLUMN` type `INTEGER` - Column number of the statement
+ - `RDB$COUNTER` type `BIGINT` - Number of executed times of the statement
+ - `RDB$MIN_TIME` type `BIGINT` - Minimal time (in nanoseconds) of a statement execution
+ - `RDB$MAX_TIME` type `BIGINT` - Maximum time (in nanoseconds) of a statement execution
+ - `RDB$ACCUMULATED_TIME` type `BIGINT` - Accumulated execution time (in nanoseconds) of the statement
```

Comment:
RDB$TOTAL_TIME maybe?

Reply:
> RDB$TOTAL_TIME maybe?

Agree.

Reply:
Should it be MIN/MAX TIME our MINIMUM/MAXIMUM?

Reply:
MIN/MAX is fine to me.

Reply:
Can you also add AVG_TIME?